src/android/FileTransferBackground.java

```diff
@@ -1,216 +1,398 @@
+        try {
+            parameters = convertToHashMap((JSONObject) payload.get("parameters"));
+        } catch (JSONException e) {
+            Log.i(TAG, "eventLabel='could not parse request headers' uploadId='" + uploadId + "' error='" + e.getMessage() + "'");
+            sendAddingUploadError(uploadId, e);
+            return;
+        }
+        final List<String> parameterNames = new ArrayList<>(parameters.size());
+        final Map<String, Object> parameterValues = new HashMap<>(parameters.size());
+        for (Map.Entry<String, Object> entry : parameters.entrySet()) {
+            parameterValues.put(UploadTask.KEY_INPUT_PARAMETER_VALUE_PREFIX + parameterNames.size(), entry.getValue());
+            parameterNames.add(entry.getKey());
+        }
+
+        startUpload(uploadId, new Data.Builder()
+                .putString(UploadTask.KEY_INPUT_ID, uploadId)
+                .putString(UploadTask.KEY_INPUT_URL, (String) payload.get("serverUrl"))
+                .putString(UploadTask.KEY_INPUT_FILEPATH, (String) payload.get("filePath"))
+                .putString(UploadTask.KEY_INPUT_FILE_KEY, (String) payload.get("fileKey"))
+
+                .putInt(UploadTask.KEY_INPUT_HEADERS_COUNT, headersNames.size())
+                .putStringArray(UploadTask.KEY_INPUT_HEADERS_NAMES, headersNames.toArray(new String[0]))
+                .putAll(headerValues)
+
+                .putInt(UploadTask.KEY_INPUT_PARAMETERS_COUNT, parameterNames.size())
+                .putStringArray(UploadTask.KEY_INPUT_PARAMETERS_NAMES, parameterNames.toArray(new String[0]))
+                .putAll(parameterValues)
+
+                .putString(UploadTask.KEY_INPUT_NOTIFICATION_TITLE, (String) payload.get("notificationTitle"))
+                .build()
+        );
+    }
+
+    private void startUpload(final String uploadId, final Data payload) {
+        Log.i(TAG, "startUpload: Starting work via work manager");
+
+        OneTimeWorkRequest workRequest = new OneTimeWorkRequest.Builder(UploadTask.class)
+                .setConstraints(new Constraints.Builder()
+                        .setRequiredNetworkType(NetworkType.UNMETERED).build()
+                )
+                .keepResultsForAtLeast(0, TimeUnit.MILLISECONDS)
+                .setBackoffCriteria(BackoffPolicy.EXPONENTIAL, 1, TimeUnit.MINUTES)
+                .addTag(WORK_TAG_UPLOAD)
+                .setInputData(payload)
+                .build();
+
+        OneTimeWorkRequest ackRequest = new OneTimeWorkRequest.Builder(PendingAcknowledgeTask.class)
+                .keepResultsForAtLeast(1, TimeUnit.DAYS)
+                .setBackoffCriteria(BackoffPolicy.LINEAR, 10, TimeUnit.SECONDS)
+                .addTag(WORK_TAG_ACK)
+                .build();
+
+        WorkManager.getInstance(cordova.getContext())
+                .beginUniqueWork(uploadId, ExistingWorkPolicy.KEEP, workRequest)
+                .then(ackRequest)
+                .enqueue();
```

Comment:
On n'a pas mieux pour gérer les Ack qu'un autre worker ? 🤔

Reply:
Not really but while it is not running (like in its enqued state), a task is just a line in a database, not so different than the previous implementation.